packages/run-protocol/src/vaultFactory/vaultDirector.js

```diff
@@ -74,260 +81,296 @@ const initState = (zcf, electorateParamManager, debtMint) => {
-    // zcf,
-  } = initState(zcf, electorateParamManager, debtMint);
+const makeVaultInvitation = ({ state }) => {
+  const { collateralTypes, zcf } = state;
 
-  /** Make a loan in the vaultManager based on the collateral type.
-   *
-   * @deprecated
-   */
-  const makeVaultInvitation = () => {
-    /** @param {ZCFSeat} seat */
-    const makeVaultHook = async seat => {
-      assertProposalShape(seat, {
-        give: { Collateral: null },
-        want: { RUN: null },
-      });
-      const {
-        give: { Collateral: collateralAmount },
-      } = seat.getProposal();
-      const { brand: brandIn } = collateralAmount;
-      assert(
-        collateralTypes.has(brandIn),
-        X`Not a supported collateral type ${brandIn}`,
-      );
-      /** @type {VaultManager} */
-      const mgr = collateralTypes.get(brandIn);
-      return mgr.makeVaultKit(seat);
-    };
-    return zcf.makeInvitation(makeVaultHook, 'MakeVault');
-  };
-
-  // TODO put on machineFacet for use again in publicFacet
-  const getCollaterals = async () => {
-    // should be collateralTypes.map((vm, brand) => ({
-    return harden(
-      Promise.all(
-        [...collateralTypes.entries()].map(async ([brand, vm]) => {
-          const priceQuote = await vm.getCollateralQuote();
-          return {
-            brand,
-            interestRate: vm.getGovernedParams().getInterestRate(),
-            liquidationMargin: vm.getGovernedParams().getLiquidationMargin(),
-            stabilityFee: vm.getGovernedParams().getLoanFee(),
-            marketPrice: makeRatioFromAmounts(
-              getAmountOut(priceQuote),
-              getAmountIn(priceQuote),
-            ),
-          };
-        }),
-      ),
+  /** @param {ZCFSeat} seat */
+  const makeVaultHook = async seat => {
+    assertProposalShape(seat, {
+      give: { Collateral: null },
+      want: { RUN: null },
+    });
+    const {
+      give: { Collateral: collateralAmount },
+    } = seat.getProposal();
+    const { brand: brandIn } = collateralAmount;
+    assert(
+      collateralTypes.has(brandIn),
+      X`Not a supported collateral type ${brandIn}`,
     );
+    /** @type {VaultManager} */
+    const mgr = collateralTypes.get(brandIn);
+    return mgr.makeVaultKit(seat);
   };
+  return zcf.makeInvitation(makeVaultHook, 'MakeVault');
+};
 
-  /** @type {VaultFactory} */
-  const machineBehavior = {
-    // TODO move this under governance #3924
-    /** @type {AddVaultType} */
-    addVaultType: async (
-      collateralIssuer,
-      collateralKeyword,
-      initialParamValues,
-    ) => {
-      await zcf.saveIssuer(collateralIssuer, collateralKeyword);
-      const collateralBrand = zcf.getBrandForIssuer(collateralIssuer);
-      // We create only one vault per collateralType.
-      assert(
-        !collateralTypes.has(collateralBrand),
-        `Collateral brand ${collateralBrand} has already been added`,
-      );
+// TODO put on machineFacet for use again in publicFacet
```

Comment:
what I went with is defining this at top scope so it can be shared.

thoughts on that vs giving it a primary facet and letting another reexport?

Reply:
both approaches seem fine where appropriate. This seems sufficient for this use. I can think of other places where we might want a facet.  (I'm considering that for the AMM's singlePool, but that has more justification to be a facet.)